src/System.Net.WebSockets.Client/tests/SendReceiveTest.cs

```diff
@@ -157,45 +157,42 @@ public class SendReceiveTest : ClientWebSocketTestBase
                 var recvBuffer = new byte[100];
                 var recvSegment = new ArraySegment<byte>(recvBuffer);
 
-                try
+                for (int i = 0; i < tasks.Length; i++)
                 {
-                    for (int i = 0; i < tasks.Length; i++)
-                    {
-                        tasks[i] = cws.ReceiveAsync(recvSegment, cts.Token);
-                    }
+                    tasks[i] = cws.ReceiveAsync(recvSegment, cts.Token);
+                }
 
-                    Task.WaitAll(tasks);
+                Exception e = Record.Exception(() => Task.WaitAll(tasks));
+
+                if (e == null)
+                {
                     Assert.Equal(WebSocketState.Open, cws.State);
                 }
-                catch (AggregateException ag)
+                else
                 {
-                    foreach (var ex in ag.InnerExceptions)
+                    AggregateException ae = Assert.IsType<AggregateException>(e);
+
+                    Assert.All(ae.InnerExceptions, ex =>
                     {
+                        Assert.True(ex is InvalidOperationException || ex is WebSocketException,
```

Comment:
> I prefer doing this

Everyone has their own personal preferences.  If one is obviously better, than a change makes sense.  Otherwise, as in this case, I'm inclined to just keep things the way they are.  Someone else could come along and think "I prefer this other way", and change it back.